
i686-pld-linux-gcc -DHAVE_CONFIG_H -I. -I../..   -DSYS_LINUX -I../../include `top_builddir="../.." ../../vlc-config --cflags builtin pic i422_yuy2_mmx` -Wsign-compare -Wall -O2 -march=i686  -DALSA_PCM_OLD_HW_PARAMS_API -pipe -MT libi422_yuy2_mmx_a-i422_yuy2.o -MD -MP -MF .deps/libi422_yuy2_mmx_a-i422_yuy2.Tpo -c -o libi422_yuy2_mmx_a-i422_yuy2.o `test -f 'i422_yuy2.c' || echo './'`i422_yuy2.c
i420_rgb16.c: In function `I420_RGB16__i420_rgb_mmx':
i420_rgb16.c:346: error: `__m64' undeclared (first use in this function)
i420_rgb16.c:346: error: (Each undeclared identifier is reported only once
i420_rgb16.c:346: error: for each function it appears in.)
i420_rgb16.c:346: error: syntax error before "mm0"
i420_rgb16.c:347: warning: unused variable `tmp64'
i420_rgb16.c: In function `I420_RGB32__i420_rgb_mmx':
i420_rgb16.c:532: error: `__m64' undeclared (first use in this function)
i420_rgb16.c:532: error: syntax error before "mm0"
i420_rgb16.c:533: warning: unused variable `tmp64'
mv -f .deps/libi420_rgb_mmx_a-i420_rgb.Tpo .deps/libi420_rgb_mmx_a-i420_rgb.Po

copy __m64 from /usr/lib/gcc-lib/i686-pld-linux/3.3.6/include/mmintrin.h

--- vlc-0.8.6/modules/video_chroma/i420_rgb16.c~	2006-12-13 01:23:37.271827185 +0200
+++ vlc-0.8.6/modules/video_chroma/i420_rgb16.c	2006-12-13 01:23:39.341873805 +0200
@@ -343,6 +343,7 @@
         {
 #if defined (MODULE_NAME_IS_i420_rgb_mmx)
 #   if defined (HAVE_MMX_INTRINSICS)
+typedef int __m64 __attribute__ ((__mode__ (__V2SI__)));
             __m64 mm0, mm1, mm2, mm3, mm4, mm5, mm6, mm7;
             uint64_t tmp64;
 #   endif
@@ -529,6 +530,7 @@
         {
 #if defined (MODULE_NAME_IS_i420_rgb_mmx)
 #   if defined (HAVE_MMX_INTRINSICS)
+typedef int __m64 __attribute__ ((__mode__ (__V2SI__)));
             __m64 mm0, mm1, mm2, mm3, mm4, mm5, mm6, mm7;
             uint64_t tmp64;
 #   endif
